Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -1075,12 +1075,27 @@ public final class PackageBuilder {
             case .library, .test:
                 break
             case .executable:
-                let executableTargets = targets.filter({ $0.type == .executable })
-                if executableTargets.count != 1 {
-                    diagnostics.emit(
-                        .invalidExecutableProductDecl(product.name),
-                        location: diagnosticLocation()
-                    )
+                let executableTargetCount = targets.filter { $0.type == .executable }.count
+                if executableTargetCount != 1 {
+                    if executableTargetCount == 0 {
+                        if let target = targets.spm_only {
+                            diagnostics.emit(
+                                .executableProductTargetNotExecutable(product: product.name, target: target.name),
+                                location: diagnosticLocation()
+                            )
+                        } else {
+                            diagnostics.emit(
+                                .executableProductWithoutExecutableTarget(product: product.name),
+                                location: diagnosticLocation()
+                            )
+                        }
+                    } else {
+                        diagnostics.emit(
+                            .executableProductWithMoreThanOneExecutableTarget(product: product.name),
+                            location: diagnosticLocation()
+                        )
+                    }
```

Comment:
Can you put this part in a separate function? Something like "diagnoseMultipleTargetsInExecutableProduct" or "validateExecutableProduct"

Reply:
Done!